lib/tasks/unpublish.rake

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+namespace :unpublish do
+  desc "Retire a document on GOV.UK e.g. unpublish:retire_document CONTENT_ID='a-content-id' NOTE='A note'"
+  task retire_document: :environment do
+    raise "Missing CONTENT_ID value" if ENV["CONTENT_ID"].nil?
+    raise "Missing NOTE value" if ENV["NOTE"].nil?
+
+    content_id = ENV["CONTENT_ID"]
+    explanatory_note = ENV["NOTE"]
+    locale = ENV["LOCALE"] || "en"
+
+    document = Document.find_by(content_id: content_id, locale: locale)
+    raise "Document must have a published version before it can be retired" unless document.has_live_version_on_govuk
+
+    DocumentUnpublishingService.new.retire(document, explanatory_note, locale: locale) if document.present?
+  end
+
+  desc "Remove a document from GOV.UK e.g. unpublish:remove_document CONTENT_ID='a-content-id'"
+  task remove_document: :environment do
+    raise "Missing CONTENT_ID value" if ENV["CONTENT_ID"].nil?
+
+    content_id = ENV["CONTENT_ID"]
+    explanatory_note = ENV["NOTE"]
+    alternative_path = ENV["NEW_PATH"]
+    locale = ENV["LOCALE"] || "en"
+
+    document = Document.find_by(content_id: content_id, locale: locale)
+    raise "Document must have a published version before it can be removed" unless document.has_live_version_on_govuk
+
+    if document.present?
+      DocumentUnpublishingService.new.remove(
+        document,
+        explanatory_note: explanatory_note,
+        alternative_path: alternative_path,
+        locale: locale,
+      )
+    end
+  end
+
+  desc "Remove and redirect a document on GOV.UK e.g. unpublish:remove_and_redirect_document CONTENT_ID='a-content-id' NEW_PATH='/redirect-to-here'"
+  task remove_and_redirect_document: :environment do
+    raise "Missing CONTENT_ID value" if ENV["CONTENT_ID"].nil?
+    raise "Missing NEW_PATH value" if ENV["NEW_PATH"].nil?
```

Comment:
might want to use unless ENV["NEW_PATH"].present? just to rule out empty strings too

Reply:
I think `blank?` might be the better option:

```
irb(main):001:0> a = ""
=> ""
irb(main):002:0> a.present?
=> false
irb(main):003:0> a.nil?
=> false
irb(main):004:0> a.blank?
=> true
irb(main):005:0> b = nil
=> nil
irb(main):006:0> b.present?
=> false
irb(main):007:0> b.nil?
=> true
irb(main):008:0> b.blank?
=> true
```

Reply:
Ah yes I agree!

Reply:
It's odd. For some reason the tests broke when I changed it to `present?`.